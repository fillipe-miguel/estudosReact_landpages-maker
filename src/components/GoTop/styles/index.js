import styled, { css } from 'styled-components';

export const GoTopRoot = styled.a`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.primaryColor};
    border-radius: 1rem;
    border: 2px solid ${theme.colors.white};
    bottom: 2rem;
    color: ${theme.colors.white};
    display: flex;
    font-size: ${theme.fonts.sizes.medium};
    height: 4rem;
    justify-content: center;
    position: fixed;
    right: 2rem;
    opacity: 0.5;
    z-index: 6;
    transition: all 0.3s ease-in-out;
    width: 4rem;

    &:hover {
      opacity: 1;
      border-color: ${theme.colors.secondaryColor};
      box-shadow: 0px 0px 8px 1px ${theme.colors.secondaryColor};
      color: ${theme.colors.secondaryColor};
      transform: scale(1.05);
      transition: all 0.3s ease-in-out;
    }

    &:hover > :first-child {
      animation: shakeUpDown 0.7s ease-in-out infinite;
    }
  `}

  @keyframes shakeUpDown {
    from {
      transform: translateY(0px);
    }
    to {
      transform: translateY(-5px);
    }
  }
`;

export default GoTopRoot;
