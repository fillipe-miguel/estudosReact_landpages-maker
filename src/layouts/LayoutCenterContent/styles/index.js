import styled, { css } from 'styled-components';

export const LayoutCenterContentRoot = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const InnerHtml = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.large};
    text-align: justify;

    & h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      text-align: center;
    }
  `}
`;

export default LayoutCenterContentRoot;
