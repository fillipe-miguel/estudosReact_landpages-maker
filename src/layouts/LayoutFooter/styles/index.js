import styled, { css } from 'styled-components';

import { TextComponentStyles } from '../../../components';

export const LayoutFooterRoot = styled.footer`
  ${({ theme }) => css`
    height: 8rem;
    text-align: center;
    background-color: ${theme.colors.white};
    border-top: 1px solid ${theme.colors.mediumGray};

    & ${TextComponentStyles.TextComponentRoot} {
      font-size: ${theme.fonts.sizes.small};
    }

    a {
      pointer-events: all !important;
      text-decoration: underline !important;
    }
  `}
`;

export default LayoutFooterRoot;
