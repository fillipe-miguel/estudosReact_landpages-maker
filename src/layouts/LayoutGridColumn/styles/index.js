import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const LayoutGridColumnRoot = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;

    gap: ${theme.spacings.xxlarge};

    & ${HeadingStyles.HeadingRoot} {
      margin-bottom: ${theme.spacings.small};
    }
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      flex-direction: column;

      & ${HeadingStyles.HeadingRoot} {
        margin-bottom: ${theme.spacings.xlarge};
      }
    }
  `}
`;

export const TextContainer = styled.div`
  ${() => css`
    max-width: 35rem;
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      min-width: auto;
      text-align: center;
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css`
    max-width: 64rem;
  `}
`;

export const Image = styled.img`
  ${() => css`
    max-width: 100%;
  `}
`;
