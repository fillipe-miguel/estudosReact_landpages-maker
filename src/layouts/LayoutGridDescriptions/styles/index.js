import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    & ${HeadingStyles.HeadingRoot} {
      position: relative;
      left: 5rem;
      margin: ${theme.spacings.medium} 0;
    }

    & ${HeadingStyles.HeadingRoot}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xxxlarge};
      left: -5rem;
      top: -2.5rem;
      transform: skew(-15deg);
    }
  `}
`;
