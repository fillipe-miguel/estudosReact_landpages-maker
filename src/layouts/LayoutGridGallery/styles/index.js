import styled, { css } from 'styled-components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(270px, 1fr));
    gap: ${theme.spacings.medium};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
    height: fit-content;

    &:hover > img {
      transition: all 0.3s ease-in-out;
      transform: scale(1.2) rotate(9deg);
    }

    img {
      pointer-events: none;
      max-width: 100%;
    }
  `}
`;
