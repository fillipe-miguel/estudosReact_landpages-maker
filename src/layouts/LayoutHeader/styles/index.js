import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';
import { SectionContainerStyles } from '../../../sections';

export const LayoutHeaderRoot = styled.header`
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;

  & > ${SectionContainerStyles.SectionContainerRoot} {
    padding-top: ${({theme}) => theme.spacings.small};
    padding-bottom: ${({theme}) => theme.spacings.small};;
  }

  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    border-bottom: 1px solid ${theme.colors.mediumGray};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme, ...props }) => css`
    @media ${theme.medias.lteMedium} {
      bottom: 0;
      height: 100vh;
      visibility: ${props.visible ? 'visibility' : 'hidden'};
      opacity: ${props.visible ? '1' : '0'};
      transition: all 0.3s ease-in-out;

      & > ${SectionContainerStyles.SectionContainerRoot} {
        display: grid;
        align-items: center;

        height: 100%;
        margin: 0;

        overflow-y: auto;
      }

      & ${HeadingStyles.HeadingRoot} {
        padding-bottom: ${theme.spacings.large};
      }
    }
  `}
`;

export const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      flex-direction: column;
      text-align: center;
      padding: ${theme.spacings.xxlarge};
    }
  `}
`;

export const ToggleMenuButton = styled.button`
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;

  display: none;

  font-size: 2.5rem;
  padding: 0.5rem;

  cursor: pointer;
  border: none;

  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      align-items: center;
    }
  `}
`;

export default LayoutHeaderRoot;
