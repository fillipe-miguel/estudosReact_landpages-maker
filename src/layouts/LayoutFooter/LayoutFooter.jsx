import P from 'prop-types';
import React from 'react';

import * as Styled from './styles';

import { TextComponent } from '../../components';
import { SectionContainer } from '../../sections';

function LayoutFooter({ html }) {
  return (
    <Styled.LayoutFooterRoot>
      <SectionContainer>
        <TextComponent asHtml>{html}</TextComponent>
      </SectionContainer>
    </Styled.LayoutFooterRoot>
  );
}

LayoutFooter.propTypes = {
  html: P.string.isRequired,
};

export default LayoutFooter;
