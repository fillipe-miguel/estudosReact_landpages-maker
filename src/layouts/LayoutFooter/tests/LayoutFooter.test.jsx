/* eslint-disable no-undef */
import { render } from '@testing-library/react';
import LayoutFooter from '..';

import { MyThemeProvider } from '../../../styles';

describe('<LayoutFooter />', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <MyThemeProvider>
        <LayoutFooter footerHtml={'<a href="#">Teste intenso aqui!</a>'} />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
