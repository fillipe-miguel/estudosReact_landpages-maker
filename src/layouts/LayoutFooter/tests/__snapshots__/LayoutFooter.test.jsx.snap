// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutFooter /> Should match snapshot 1`] = `
.c3 {
  font-size: 2.4rem;
  color: #0A1128;
}

.c0 {
  height: 8rem;
  text-align: center;
  background-color: #FFF;
  border-top: 1px solid #DDDDDD;
}

.c0 .c2 {
  font-size: 1.6rem;
}

.c0 a {
  pointer-events: all !important;
  -webkit-text-decoration: underline !important;
  text-decoration: underline !important;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    >
      <a
        href="#"
      >
        Teste intenso aqui!
      </a>
    </div>
  </div>
</footer>
`;
